src/stan-users-guide/proportionality-constants.Rmd

```diff
@@ -0,0 +1,150 @@
+to select between them is by determining if the proportionality constants are
+necessary.
+
+The first two syntaxes drop proportionality constants:
+
+```
+x ~ normal(0, 1);
+target += normal_lupdf(x | 0, 1);
+```
+
+The final syntax computes the full log density (dropping no constants):
+
+```
+target += normal_lpdf(x | 0, 1);
+```
+
+For discrete distributions, the `target +=` syntax is `_lupmf` and `_lpmf`
+instead:
+
+```
+y ~ bernoulli(0.5);
+target += bernoulli_lupmf(y | 0.5);
+target += bernoulli_lpmf(y | 0.5);
+```
+
+## Dropping Proportionality Constants
+
+If a density $p(\theta)$ can be factored into $K g(\theta)$ where $K$ are all
+the factors that are a not a function of $\theta$ and $g(\theta)$ are all the
+terms that are a function of $\theta$, then it is said that $g(\theta)$
+is proportional to $p(\theta)$ up to a constant.
+
+The advantage of all this is that sometimes $K$ is expensive to compute and if
+it is not a function of the distribution that is to be sampled (or optimized or
+approximated with variational inference), there is no need to compute it
+because it will not affect the results.
+
+Stan takes advantage of the proportionality constant fact with the `~` syntax.
+Take for instance the normal likelihood:
+
+```
+data {
+  real mu;
+  real<lower = 0.0> sigma;
+}
+parameters {
+  real x;
+}
+model {
+  x ~ normal(mu, sigma);
+}
+```
+
+Syntactically, this is just shorthand for the equivalent model that replaces the
+`~` syntax with a `target +=` statement and a `normal_lupdf` function call:
+
+```
+data {
+  real mu;
+  real<lower = 0.0> sigma;
+}
+parameters {
+  real x;
+}
+model {
+  target += normal_lupdf(x | mu, sigma)
+}
+```
+
+The function `normal_lupdf` is only guaranteed to return the log density of the
+normal distribution up to a proportionality constant density to be sampled. The
+proportionality constant itself is not defined. The full log density of the
+statement here is:
+
+$$
+\textsf{normal_lupdf}(x | \mu, \sigma) =
+-\log \left( \sigma \sqrt{2 \pi} \right)
+-\frac{1}{2} \left( \frac{x - \mu}{\sigma} \right)^2
+$$
+
+Now because the density here is only a function of $x$, the additive terms in
+the log density that are not a function of $x$ can be dropped. In this case it
+is enough to know only the quadratic term:
+
+$$
+\textsf{normal_lupdf}(x | \mu, \sigma) \propto
+-\frac{1}{2} \left( \frac{x - \mu}{\sigma} \right)^2
+$$
+
+## Keeping Proportionality Constants
+
+In the case that the proportionality constants were needed for a normal log
+density the function `normal_lpdf` can be used. For clarity, if there is ever
+a situation where it is unclear if the normalization is necessary, it should
+always be safe to include it. Only use the `~` or `target += normal_lupdf`
+syntaxes if it is absolutely clear that the proportionality constants are not
+necessary.
+
+## User-defined Distributions
+
+While it is possible to define custom distributions in Stan, it is not yet
```

Comment:
Not sure we need to say "not yet". No one is currently thinking about/working on allowing these. We might end up adding it soon, but if we don't this will be a dangling "soon" :)

Reply:
Just mention that currently you always define a UDF as normalized but can call it unnormalized. Calling it as unnormalized will however only help with any calls to Stan-defined lpdfs/lpmfs.